### PR TITLE
Zap updates to resolve zap warnings for lock, thermostat , window and light-switch examples.

### PR DESCRIPTION
**Pull request by Sarthak-Shaha**
Problem:
Latest Zap shows error warning while building for the clusters / commands / attributes which are not up to the latest specifications. The zap files for lock app, thermostat , light-switch and silabs lighting are not up to the spec. 

Fix / solution:
1. Modifies the zap files for lock app, thermostat , light-switch and silabs lighting to comply with latest spec
2. modifies the xml files to reflect the changes in mandatory commands
3. generated the .matter files for zap changes

Made following changes
For all zap files
a. enabled cluster: Diagnostic Logs server : RetrieveLogsResponse command

1. Light- switch :
    a. Identify client : enabled identify outgoing commands
2. Thermostat
    a. cluster: Identify, mandatory attribute: Identify Type.
3. Silabs Lighting app thread/wifi:
    TODO : Need to enable scenes cluster in Matter_1.3

### CODE REVIEW

**Comment by andy31415** [comment]
@Sarthak-Shaha it looks like ESP32 fails with `/__w/connectedhomeip/connectedhomeip/examples/lock-app/esp32/third_party/connectedhomeip/src/app/util/util.cpp:109: undefined reference to `_Z52MatterTimeFormatLocalizationPluginServerInitCallbackv'`

File: examples/lock-app/lock-common/lock-app.matter [hunk 1]
```diff
@@ -427,6 +427,50 @@ cluster LocalizationConfiguration = 43 {
   readonly attribute int16u clusterRevision = 65533;
 }
 
+/** Nodes should be expected to be deployed to any and all regions of the world. These global regions
+      may have differing preferences for how dates and times are conveyed. As such, Nodes that visually
+      or audibly convey time information need a mechanism by which they can be configured to use a
+      user’s preferred format. */
+cluster TimeFormatLocalization = 44 {
+  revision 1; // NOTE: Default/not specifically set
```
Comment (andy31415) [inline examples/lock-app/lock-common/lock-app.matter:435]: this seems to indicate that we need the XML to define the global attribute value (so we do not have default here)

**Comment by mkardous-silabs** [comment]
I don't see any ICDM cluster changes in the lock zap file but the `RegisterClientResponse` is not mandatory. It is only required if the Check-In protocol featuremap is 1 which it shouldn't on the lock app.

File: examples/light-switch-app/light-switch-common/light-switch-app.matter [hunk 2]
```diff
@@ -2563,6 +2589,15 @@ endpoint 0 {
     ram      attribute clusterRevision default = 1;
   }
 
+  server cluster UnitLocalization {
```
Comment (bzbarsky-apple) [inline examples/light-switch-app/light-switch-common/light-switch-app.matter:2592]: Why is this here?  It shouldn't need to be here.  If ZAP says it does, ZAP is wrong.

File: examples/lock-app/lock-common/lock-app.matter [hunk 3]
```diff
@@ -2565,6 +2609,16 @@ endpoint 0 {
     ram      attribute clusterRevision default = 1;
   }
 
+  server cluster TimeFormatLocalization {
```
Comment (bzbarsky-apple) [inline examples/lock-app/lock-common/lock-app.matter:2612]: Why is this here?  It's not required by spec.

File: examples/thermostat/thermostat-common/thermostat.matter [hunk 4]
```diff
@@ -2359,6 +2361,8 @@ endpoint 0 {
     callback attribute timeSinceReset;
     ram      attribute featureMap default = 3;
     ram      attribute clusterRevision default = 1;
+
+    handle command ResetCounts;
```
Comment (bzbarsky-apple) [inline examples/thermostat/thermostat-common/thermostat.matter:2365]: Why is this being added?  This is an optional command.

File: examples/window-app/common/window-app.matter [hunk 5]
```diff
@@ -2283,6 +2283,8 @@ endpoint 0 {
     callback attribute timeSinceReset;
     ram      attribute featureMap default = 3;
     ram      attribute clusterRevision default = 1;
+
+    handle command ResetCounts;
```
Comment (bzbarsky-apple) [inline examples/window-app/common/window-app.matter:2287]: This is an optional command.

File: examples/window-app/common/window-app.matter [hunk 6]
```diff
@@ -2462,6 +2464,7 @@ endpoint 2 {
     ram      attribute clusterRevision default = 4;
 
     handle command Identify;
+    handle command TriggerEffect;
```
Comment (bzbarsky-apple) [inline examples/window-app/common/window-app.matter:2467]: This is an optional command.

File: src/app/zap-templates/zcl/data-model/chip/identify-cluster.xml [hunk 7]
```diff
@@ -62,7 +62,7 @@ limitations under the License.
       <arg name="IdentifyTime" type="int16u"/>
       <access op="invoke" privilege="manage"/>
     </command>
-    <command source="client" code="0x40" name="TriggerEffect" optional="true">
+    <command source="client" code="0x40" name="TriggerEffect" optional="false">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/identify-cluster.xml:65]: This is _an optional command_ in the spec.  Why is this being changed to not follow the spec?

File: src/app/zap-templates/zcl/data-model/chip/time-format-localization-cluster.xml [hunk 8]
```diff
@@ -55,6 +55,7 @@ limitations under the License.
       may have differing preferences for how dates and times are conveyed. As such, Nodes that visually
       or audibly convey time information need a mechanism by which they can be configured to use a
       user’s preferred format.</description>
+    <globalAttribute side="either" code="0xFFFD" value="2"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/time-format-localization-cluster.xml:58]: Why is this change here?  Spec has 1 as the cluster revision for this cluster, as far as I can tell.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/window-app/common/window-app.matter (hunk 6 above)
Comment (brdandu) [reply examples/window-app/common/window-app.matter:2467]: This might be related to the xml change which would make it mandatory
`<command source="client" code="0x40" name="TriggerEffect" optional="true">`

File: examples/light-switch-app/light-switch-common/light-switch-app.matter (hunk 2 above)
Comment (brdandu) [reply examples/light-switch-app/light-switch-common/light-switch-app.matter:2592]: `<include cluster="Unit Localization" client="false" server="true" clientLocked="true" serverLocked="false"></include>`
By default the XML says that this cluster is true with `server="true"` .That is why ZAP is warning the user. Should this be turned to false by default?

Comment (brdandu) [reply examples/light-switch-app/light-switch-common/light-switch-app.matter:2592]: Based on the user, this can go either way with whether the user needs Unit localization or not because by default it is expected to be enabled.

File: examples/thermostat/thermostat-common/thermostat.matter (hunk 4 above)
Comment (brdandu) [reply examples/thermostat/thermostat-common/thermostat.matter:2365]: `<command source="client" code="0x00" name="ResetCounts" optional="false" cli="chip ethernet_network_diagnostics resetcounts">
      <description>Reception of this command SHALL reset the attributes: PacketRxCount, PacketTxCount, TxErrCount, CollisionCount, OverrunCount to 0</description>
      <access op="invoke" role="manage"/>
    </command>`

Ethernet Network Diagnostics cluster does not say that this cluster is optional. `optional=false`

File: examples/lock-app/lock-common/lock-app.matter (hunk 3 above)
Comment (brdandu) [reply examples/lock-app/lock-common/lock-app.matter:2612]: Matter root device has this enabled by default:
```<include cluster="Time Format Localization" client="false" server="true" clientLocked="true" serverLocked="false">
                <requireAttribute>HOUR_FORMAT</requireAttribute>
            </include>```

File: examples/window-app/common/window-app.matter (hunk 5 above)
Comment (brdandu) [reply examples/window-app/common/window-app.matter:2287]: <command source="client" code="0x00" name="ResetCounts" optional="false" cli="chip ethernet_network_diagnostics resetcounts"> <description>Reception of this command SHALL reset the attributes: PacketRxCount, PacketTxCount, TxErrCount, CollisionCount, OverrunCount to 0</description> <access op="invoke" role="manage"/> </command>

Ethernet Network Diagnostics cluster does not say that this cluster is optional. optional=false

File: src/app/zap-templates/zcl/data-model/chip/identify-cluster.xml (hunk 7 above)
Comment (brdandu) [reply src/app/zap-templates/zcl/data-model/chip/identify-cluster.xml:65]: This should be reverted and that should also revert the above TriggerEffect code change

File: src/app/zap-templates/zcl/data-model/chip/time-format-localization-cluster.xml (hunk 8 above)
Comment (brdandu) [reply src/app/zap-templates/zcl/data-model/chip/time-format-localization-cluster.xml:58]: Not sure why this was added here

File: examples/lock-app/lock-common/lock-app.matter (hunk 3 above)
Comment (bzbarsky-apple) [reply examples/lock-app/lock-common/lock-app.matter:2612]: Well, then that's wrong in the XML and needs to be fixed.

File: examples/thermostat/thermostat-common/thermostat.matter (hunk 4 above)
Comment (bzbarsky-apple) [reply examples/thermostat/thermostat-common/thermostat.matter:2365]: The spec says it's optional.

Comment (bzbarsky-apple) [reply examples/thermostat/thermostat-common/thermostat.matter:2365]: https://github.com/project-chip/connectedhomeip/pull/31721 to fix the broken XML.

**Comment by selissia** [comment]
Note https://github.com/project-chip/connectedhomeip/pull/32170 which has various XML changes.

**Comment by Sarthak-Shaha** [comment]
> I don't see any ICDM cluster changes in the lock zap file but the `RegisterClientResponse` is not mandatory. It is only required if the Check-In protocol featuremap is 1 which it shouldn't on the lock app.

File: examples/lighting-app/silabs/data_model/lighting-thread-app.zap [hunk 9]
```diff
@@ -1509,6 +1509,14 @@
               "source": "client",
               "isIncoming": 1,
               "isEnabled": 1
+            },
+            {
+              "name": "RetrieveLogsResponse",
```
Comment (jmartinez-silabs) [inline examples/lighting-app/silabs/data_model/lighting-thread-app.zap:1514]: Diagnostic logs are not currently supported on Silabs platform. since this zap config is specific to the Silabs platform we should remove this cluster from it.

The other zap files are common zap config. We currently have to live it it enabled but not this one

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 10]
```diff
@@ -74,10 +74,6 @@ limitations under the License.
                 <requireAttribute>ACTIVE_LOCALE</requireAttribute>
                 <requireAttribute>SUPPORTED_LOCALES</requireAttribute>
             </include>
-            <include cluster="Time Format Localization" client="false" server="true" clientLocked="true" serverLocked="false">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:77]: Why is this being removed?

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 11]
```diff
@@ -74,10 +74,6 @@ limitations under the License.
...
-                <requireAttribute>HOUR_FORMAT</requireAttribute>
-            </include>
-            <include cluster="Unit Localization" client="false" server="true" clientLocked="true" serverLocked="false"></include>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:80]: Why is this being removed?

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 10 above)
Comment (jmartinez-silabs) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:77]: The Cluster shall indeed stay listed under root endpoint but it is not mandatory so only change server="true" to server="false" and not require any attribute should do what you are looking for. Same for Unit Localization.
